tests/conversion

```diff
@@ -0,0 +1,264 @@
+
+        # Install rsync and lxd-migrate.
+        lxc exec "${instName}" -- apt-get update
+        lxc exec "${instName}" -- apt-get install --no-install-recommends -y rsync file
+        lxc exec "${instName}" -- snap install go --classic
+        lxc exec "${instName}" -- go install github.com/canonical/lxd/lxd-migrate@latest
+
+        # Set instName to the name of the new container that will be created
+        # from the existing one.
+        instName="${instName}-migration"
+    else
+        echo "Invalid instance type '${instType}'. Valid types are 'container' and 'vm'."
+        return 1
+    fi
+
+    # Generate trust token for conversion.
+    token="$(lxc config trust add --quiet --name migrate)"
+    if [ -z "${token}" ]; then
+        echo "Failed to generate LXD trust token!"
+        return 1
+    fi
+
+    # Migration questions.
+    {
+        if [ "${instType}" = "vm" ]; then
+            echo "n"           # Do not use local unix socket.
+        fi
+
+        echo "${hostAddr}"     # Address of the target LXD server.
+        sleep 1
+        echo "y"               # Yes, this is correct fingerprint.
+        sleep 1
+        echo "1"               # Use a certificate token.
+        echo "${token}"        # Token.
+        echo "${instTypeCode}" # Instace type (1 == container, 2 == virtual-machine).
+
+        if [ "$(lxc project ls -f csv | wc -l)" -gt 1 ]; then
+            echo "default"     # Project name (required if there is more then 1 project)
+        fi
+
+        echo "${instName}"     # Instance name.
+        echo "${imgPath}"      # Local image path (or filesystem path in case of container).
+        echo "${uefi_sb}"      # Enable UEFI secure boot.
+
+        # Configure storage pool.
+        if [ "${poolName}" != "" ]; then
+            echo "4"           # Change storage pool settings.
+            echo "${poolName}" # Pool name.
+            echo "yes"         # Configure pool size?
+            echo "10GiB"       # Pool size.
+        fi
+
+        # Begin the migration with the above configuration
+        sleep 1
+        echo "1"
+    } | $lxdMigrateCmd
+
+    # Start the instance.
+    echo "Starting instance ${instName}..."
+    lxc start "${instName}"
+
+    # Wait for the instance to be ready and ensure it has global ip address.
+    echo "Waiting instance ${instName} to start..."
+    waitInstanceBooted "${instName}"
+    sleep 10
+
+    # Print the instances.
+    lxc list
+
+    echo "Check network connectivity of instance ${instName}..."
+    checkIPAddresses "${instName}"
+
+    echo "Check DNS resolution of instance ${instName}..."
+    checkDNS "${instName}"
+
+    echo "Stopping instance ${instName}..."
+    lxc stop "${instName}" --timeout=30
+
+    # Cleanup.
+    lxc delete -f "${instName}"
+
+    if [ "${poolName}" != "" ]; then
+        lxc storage delete "${poolName}"
+    fi
+}
+
+# Test container migration using conversion mode.
```

Comment:
Why is this check not wrapped in the hasNeededAPIExtension check below, yet it mentions "conversion mode"?

Reply:
Because `lxd-migrate` has to fallback to migration if conversion is not available. For virtual machines it will fallback as well, but the error for "unsupported" format will be thrown, therefore, I've limited it to run VM tests only when conversion is enabled, as we are not testing raw images

Reply:
Could you add a comment explaining that, because "conversion mode" is unclear here.